subsys/bluetooth/host/ecc.c

```diff
@@ -207,7 +207,7 @@ void bt_hci_evt_le_pkey_complete(struct net_buf *buf)
 	struct bt_hci_evt_le_p256_public_key_complete *evt = (void *)buf->data;
 	struct bt_pub_key_cb *cb;
 
-	LOG_DBG("status: 0x%02x", evt->status);
+	LOG_DBG("status: %s(0x%02x)", bt_hci_err_to_str(evt->status), evt->status);
```

Comment:
Did we ever consider making the functions like `bt_hci_err_to_str` similar to `bt_addr_le_to_str`? i.e. it returns a value consisting of both a static string, and a variable value so that the string returned by `bt_hci_err_to_str` is always something like `BT_HCI_ERR_AUTH_FAIL(0x05)` so we don't need to add this type of formatting everywhere?

Reply:
And of course in the case of `CONFIG_BT_DEBUG_HCI_ERR_TO_STR=n` it would just return `0x05`

Reply:
> Did we ever consider making the functions like bt_hci_err_to_str similar to bt_addr_le_to_str? 
No I didn't. This sounds like a much better approach. Thanks for the suggestion!

Reply:
Just to check if I've understood your proposal, here is how it would look:

```c
LOG_DBG("Status is: %s", bt_hci_err_to_str(status));
...

#ifdef CONFIG_BT_DEBUG_HCI_ERR_TO_STR
const char *bt_hci_err_to_str(uint8_t err)
{
  /* Existing implementation */
}
#else
const char *bt_hci_err_to_str(uint8_t err)
{
  static char err_as_hex[5];
  sprintf(err_as_hex, "0x%02x", err);
  return err_as_hex;
}
#endif
```

This will add some overhead, but probably not too much.

From what I've understood, the logging subsystem handles this nicely.

Reply:
It would all have been simple if we converted a constant to a string instead of a variable. I agree that this is worse that we have today. I'm not sure how to make the tradeoff here

Reply:
Indeed. It would have been trivial to do with the function was along the lines of 
```c
const char *bt_hci_err_to_str(uint8_t hci_err)
{
	#define HCI_ERR(err) [err] = #err

	const char * const mapping_table[] = {
		[BT_HCI_ERR_SUCCESS] = 
#if CONFIG_BT_DEBUG_HCI_ERR_TO_STR
			HCI_ERR(BT_HCI_ERR_SUCCESS)
#endif
			"(0x00)",
		[BT_HCI_ERR_UNKNOWN_CMD] = 
#if CONFIG_BT_DEBUG_HCI_ERR_TO_STR
			"BT_HCI_ERR_UNKNOWN_CMD"
#endif
			"(0x01)",
		...
}
```

So that `bt_hci_err_to_str` would either return the value or the full string